src/control/server/config.go

```diff
@@ -242,42 +265,40 @@ func (c *configuration) cmdlineOverride(opts *cliOptions) {
 	if opts.Port > 0 {
 		c.Port = int(opts.Port)
 	}
+	if opts.Rank != nil {
+		// global rank parameter should only apply to first server
+		c.Servers[0].Rank = opts.Rank
+	}
+
 	// override each per-server config
 	for i := range c.Servers {
+		srv := &c.Servers[i]
+
 		if opts.MountPath != "" {
 			// override each per-server config in addition to global value
 			c.ScmMountPath = opts.MountPath
-			c.Servers[i].ScmMount = opts.MountPath
-		} else if c.Servers[i].ScmMount == "" {
+			srv.ScmMount = opts.MountPath
+		} else if srv.ScmMount == "" {
 			// if scm not specified for server, apply global
-			c.Servers[i].ScmMount = c.ScmMountPath
+			srv.ScmMount = c.ScmMountPath
 		}
 		if opts.Cores > 0 {
-			c.Servers[i].Cpus, _ = setNumCores(int(opts.Cores))
+			fmt.Println("-c option deprecated, please use -t instead")
+			srv.Targets, _ = setNumCores(int(opts.Cores))
 		}
 		// Targets should override Cores if specified in cmdline or
 		// config file.
 		if opts.Targets > 0 {
-			c.Servers[i].Cpus, _ = setNumCores(opts.Targets)
-		} else if c.Targets > 0 {
-			c.Servers[i].Cpus, _ = setNumCores(c.Targets)
+			srv.Targets, _ = setNumCores(int(opts.Targets))
 		}
-		if opts.Rank != nil {
-			// override first per-server config (doesn't make sense
-			// to reply to more than one server)
-			c.Servers[0].Rank = opts.Rank
+		if *opts.NrXsHelpers != 2 {
```

Comment:
Not sure I totally get this check - This is the default, right? Any reason we wouldn't want to set srv.NrXsHelpers to this?

Reply:
because that would override a setting from the config file